src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -465,6 +471,37 @@ private string GetValue(NodeModel node)
             return data;
         }
 
+        private string GenerateWatchData(List<ProtoCore.Mirror.MirrorData> list, int t)
+        {
+            var tab = new string('\t', t);
+            var format = "{0}[{1}] {2}{3}";
+            var data = new StringBuilder(list.Count == 0 ? "Empty List" : "List: \n");
+            var i = 0;
+
+            foreach (var mirrorData in list)
+            {
+                if (mirrorData.IsCollection)
+                {
+                    data.Append(string.Format(format, tab, i++, GenerateWatchData(mirrorData.GetElements(), ++t), ""));
+                }
+                else if(mirrorData.Data != null)
+                {
+                    double number;
+                    data.Append(double.TryParse(mirrorData.StringData, out number)
+                        ? string.Format(format, tab, i++, number.ToString("F4"), "\n")
+                        : string.Format(format, tab, i++, mirrorData.Data, "\n"));
+                }
+                else
+                {
+                    data.Append(mirrorData.Class == null 
+                        ? string.Format(format, tab, i++, "null", "\n")
+                        : string.Format(format, tab, i++, mirrorData.Class.ClassName, "\n"));
+                }
+            }
+
+            return data.ToString();
+        }
+
```

Comment:
Hi @Benglin,
Yes, you are right, JSON is way better here :) I've changed the code.
Hi @pboyer,
Could you please tell me how much "similar" this node should be implemented in Flood? Should it be exactly the same node as in Dynamo (with tree view)? Or should it just represent result as a formatted text?